### ISSUE TITLE
t.w.p.j.xmlstream.toResponse fails on stanzas without id attribute

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2773]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2773|
|Type|enhancement|
|Created|2007-08-03 13:27:19Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2773 2773
type__enhancement enhancement
reporter__ralphm ralphm
priority__highest highest
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1186147639000000 1186147639000000
changetime__1186497639000000 1186497639000000
version__None None
owner__ 
cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm removed owner|
|-|-|

Up for review in [branches/toResponse-2773](branches/toResponse-2773)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @ralphm|
|-|-|

I think that the usual cleanups have been done in other branches. It looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set status to `closed`|
|-|-|

(In [20984]) Fix toResponse for XMPP stanzas without an id attribute.

Author: ralphm
Reviewer: therve
Fixes [#2773](https://github.com/twisted/twisted/issues/2773).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
